src/router/routes/tool/api/createp1p1frompack.ts

```diff
@@ -0,0 +1,95 @@
+import Joi from 'joi';
+
+import Cube from '../../../../dynamo/models/cube';
+import p1p1PackModel from '../../../../dynamo/models/p1p1Pack';
+import { csrfProtection, ensureAuth } from '../../../../routes/middleware';
+import { createHydratedP1P1Pack } from '../../../../server/util/userUtil';
+import { Request, Response } from '../../../../types/express';
+import { cardFromId } from '../../../../util/carddb';
+import { isCubeViewable } from '../../../../util/cubefn';
+import { bodyValidation } from '../../../middleware/bodyValidation';
+
+const CreateP1P1FromPackSchema = Joi.object({
+  cubeId: Joi.string().uuid().required().messages({
+    'string.guid': 'Cube ID must be a valid UUID',
+    'any.required': 'Cube ID is required'
+  }),
+  seed: Joi.string().required().messages({
+    'any.required': 'Seed is required'
+  }),
+  cardIds: Joi.array()
+    .items(Joi.string().min(1))
+    .min(1)
+    .max(30)
+    .required()
+    .messages({
+      'array.min': 'At least one card ID is required',
+      'array.max': 'Pack size too large (max 30 cards)',
+      'any.required': 'Card IDs array is required'
+    })
+});
+
+export const createP1P1FromPackHandler = async (req: Request, res: Response) => {
+  try {
+    const { cubeId, seed, cardIds } = req.body;
+    const { user } = req;
+
+    // ensureAuth middleware guarantees user exists
+    if (!user) {
```

Comment:
ensureAuth redirects if the user isn't logged in, so this condition will never be triggered. So could remove to simplify